tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java

```diff
@@ -2620,7 +2623,55 @@ else if(questionStatus == Status.FAILED)
 							UIOutput.make(tableRow, "questionShowPoll", String.valueOf(i.getAttribute("questionShowPoll")));
 						}
 					}
-				}  else {
+				} else if(i.getType() == SimplePageItem.TWITTER) {
+					UIOutput.make(tableRow, "twitterSpan");
+					if (canSeeAll) {
+						String itemGroupString = simplePageBean.getItemGroupString(i, null, true);
+						String itemGroupTitles = simplePageBean.getItemGroupTitles(itemGroupString, i);
+						if (itemGroupTitles != null) {
+							itemGroupTitles = "[" + itemGroupTitles + "]";
+						}
+						UIOutput.make(tableRow, "item-groups-titles-text", itemGroupTitles);
+					}
+					if(canSeeAll || simplePageBean.isItemAvailable(i)) {
+						UIVerbatim.make(tableRow, "content", (i.getHtml() == null ? "" : i.getHtml()));
+						//Setting default values for the twitter variables
+						String tweetLimit = "5";
+						String height = "";
+						String username = "";
+						//Check if the html is neither empty nor null
+						if(StringUtils.isNotBlank(i.getHtml())){
+							//check if the html contains the variables which we expect to extract
+							if(i.getHtml().contains("data-tweet-limit") && i.getHtml().contains("data-screen-name") 
+									&& i.getHtml().contains("height") ){
+								//Extract tweet number variable from the html
+								String[] stringArray = i.getHtml().split("data-tweet-limit=");
+								//splitting the remaining text using space as delimiter
+								String[] twitterVariables = stringArray[1].split("\\s+");
+								tweetLimit = twitterVariables[0];
+								height = twitterVariables[2].split("=")[1];
```

Comment:
Although you do more validation on the initial string you want the validation to be much closer to the problems. So you want to check that you're not going to get an `ArrayOutOfBoundsException` no matter when the HTML is.